pubsub/include/common.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2024 General Motors GTO LLC
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef PUBSUB_COMMON_H
+#define PUBSUB_COMMON_H
+
+#include <up-core-api/uri.pb.h>
+#include <up-cpp/uri/builder/BuildUUri.h>
+#include <up-cpp/uri/builder/BuildUAuthority.h>
+#include <up-cpp/uri/builder/BuildEntity.h>
+#include <up-cpp/uri/builder/BuildUResource.h>
+
+uprotocol::v1::UUri getUri(
+        std::string_view const entName, int const entId, int const entMajorVers,
+        std::string_view const resName, int const resId) {
+
+    using namespace uprotocol::uri;
+
+    return BuildUUri()
+        .setAutority(BuildUAuthority().build())
```

Comment:
I might need some history here.  Is this a known misspelled interface that we accept or is there another PR to address the misspelling of setAuthority.  I looked at the code, and the interface exists and error messages within the function spell Authority (with the h).  So I'm a little confused.

Reply:
Yes, unfortunately "Authority" is misspelled "Autority" in several places. We have an open issue to find and fix that: https://github.com/eclipse-uprotocol/up-cpp/issues/89